src/main/java/de/tum/in/www1/artemis/service/hestia/ProgrammingExerciseTaskService.java

```diff
@@ -112,7 +112,9 @@ public List<ProgrammingExerciseTask> getSortedTasks(ProgrammingExercise exercise
         var unsortedTasks = programmingExerciseTaskRepository.findByExerciseIdWithTestCases(exercise.getId());
         var sortedExtractedTasks = extractTasks(exercise);
         return sortedExtractedTasks.stream()
-                .map(extractedTask -> unsortedTasks.stream().filter(task -> task.getTestCases().equals(extractedTask.getTestCases())).findFirst().orElse(null))
+                .map(extractedTask -> unsortedTasks.stream()
+                        .filter(task -> task.getTaskName().equals(extractedTask.getTaskName()) && task.getTestCases().equals(extractedTask.getTestCases())).findFirst()
```

Comment:
This is the exact same logic as the equals method in ProgrammingExerciseTask.java

Reply:
It is not. The equals method also compares the id by calling super.equals.